enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,303 @@
+This proposal does not permanently disable the strategy - rather it discourages its use by requiring admins to enable a feature set.
+This will be documented in all places where Jenkins is referenced.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, previously enabled feature sets will remain enabled.
+The bootstrap logic will ensure that clusters which rely on Jenkins pipeline builds continue to function, unless that cluster has enabled a separate feature set.
+
+On downgrade, clusters with the LegacyJenkinsPipeline feature set will continue to function, but will not be upgradeable because the feature set is not known.
+Admins will need to manually reset clusters to use the default feature set.
+
+### Version Skew Strategy
+
+Jenkins pipeline strategy builds will not be gated by the feature set until openshift-controller-manager is fully upgraded, reads the new controller configuration, and takes over leader election.
+Until that happens, Jenkins pipeline builds should be able to run as expected.
+
+## Implementation History
+
+2021-04-23: Provisional enhancement draft
+2021-07-16: Update enhancement to disable via feature gates
+
+## Drawbacks
+
+Removing Jenkins from the payload will help with the mainenance burden, because Jenkins fixes will not need to be backported to each OCP release.
+However, as long as the Jenkins pipelie strategy exists, the sync plugin will need to be maintained and updated.
+This sync plugin has been a longstanding source of bugs, and requires continued updates with each OpenShift release.
+
+This proposal aims to disable behavior that previously functions by default.
+Jenkins remains a critical part of many customer development processes - at one point Jenkins was the most widely downloaded container image from the Red Hat container catalog.
+Requriring clusters to enable a feature set could spark confusion and concern amongst end customers.
+
+Using a single supported feature set means that customers cannot enable other supported feature sets.
+For instance, customers that want to run Jenkins pipeline strategy builds cannot also enable the `LatencySensitive` feature set.
+Customers could work around this with their own custom feature set, but doing so prevents clusters from upgrading.
+
+## Alternatives
+
+Similar to the `Drawbacks` section the `Alternatives` section is used to
+highlight and record other possible approaches to delivering the value proposed
+by an enhancement.
+
+### Completely disable the Jenkins Pipeline build strategy
+
+The first draft of this proposal sought to disable the Jenkins Pipeline build strategy in its entirety.
+Not only would we disable the Jenkins pipeline strategy, we would also remove Jenkins from the OCP payload and remove a lot of functionality from the Jenkins sync plugin.
+This proved untenable in the near term because we did not have sufficient telemetry to determine the full impact of this revocation.
```

Comment:
since pipelines are already deprecated, i'm a little unclear on the value of feature-gating them but still letting people turn them on..... I guess the idea is that it puts one more barrier in front of people who are looking to use them, where we can make them aware they are opting into using a deprecated feature?

if so, we can ensure the feature gate makes it clear they are enabling a deprecated function?

I assume removal of the feature entirely is still the long term goal, that should probably be stated somewhere along w/ some indication of how we are going to decide when we can do that (e.g. telemetry)

Reply:
> if so, we can ensure the feature gate makes it clear they are enabling a deprecated function?

The name I proposed for the feature set is `LegacyJenkinsPipeline` - perhaps using `DeprecatedJenkinsPipeline` would be a stronger name?

Reply:
yes that name seems better/clearer